applications/workspaces/server/workspaces/views/api/rest_api_views.py

```diff
@@ -29,6 +30,28 @@ def get(self, id_):
                     r.update({"origin": json.loads(r.get("origin"))})
             else:
                 workspace.update({"resources": []})
+        # check if there are running import tasks
+        logger.debug("Post get, check workflows for workspace %....", workspace.get("id"))
+        workflows = argo.get_workflows(status="Running", limit=9999)
+        if workflows and workflows.items:
+            for workflow in workflows.items:
+                try:
+                    if workflow.status == "Running" and workflow.raw.spec.templates[0].metadata.labels.get(
+                        "workspace"
+                    ).strip() == str(workspace["id"]):
+                        fake_path = f"Importing resources, progress {workflow.raw.status.progress}".replace("/", " of ")
+                        workspace["resources"].append(
+                            {
+                                "id": -1,
+                                "name": "Importing resources into workspace",
+                                "origin": {"path": fake_path},
+                                "resource_type": "e",
+                                "workspace_id": workspace["id"],
+                            }
+                        )
+                    break
+                except Exception as e:
+                    pass
```

Comment:
Some logging?

Reply:
I think it will not add something... if there is an error e.g. a running workflow (not an import workflow) can fail but we are not interested in that

Reply:
I'll put a comment